packages/types/src/chain-adapters/ethereum.ts

```diff
@@ -32,10 +32,21 @@ export type QuoteFeeData = {
 }
 
 export type BuildTxInput = {
-  gasPrice: string
   gasLimit: string
   erc20ContractAddress?: string
-}
+} & (
```

Comment:
We could do something like ```
} & ({
  gasPrice: string
  maxFeePerGas?: never
  maxPriorityFeePerGas?: never
} | {
  gasPrice?: string
  maxFeePerGas: string
  maxPriorityFeePerGas: string
})
```
but I think that'll make eslint even less happy. I suppose we could make it required in both, but I'm not a fan of plumbing through the extra data because at some point we'll need to go back and prove that it's not required in order to refactor this. HDwallet took this sort of approach regarding signing BTC transactions and it made refactoring hell.

IMO input types should always be as tight as possible, and if eslint isn't happy we should address that issue separately; do you have an example of what's happening?

Reply:
I generally agree types should be as tight as possible, but I also think readability is important.

I was just thinking of having `gasPrice` required so we can always guarantee a fallback if no `maxFeePerGas` or `maxPriorityFeePerGas` is provided. A bit better than everything being optional and resulting in a throw for a user to know how to use the function call.

Reply:
That's the beauty of strict input typing! It won't throw, and there don't even need to be any runtime checks, because it won't compile in the first place.

You do have a broader point about readability. I've been looking at the [`type-fest`](https://github.com/sindresorhus/type-fest) library and thinking it would be really useful for making our type definitions more readable without compromising on specificity. What would you think about e.g.
```
export type BuildTxInput = {
  gasLimit: string
  erc20ContractAddress?: string
} & MergeExclusive<{
  gasPrice: string
}, {
  maxFeePerGas: string
  maxPriorityFeePerGas: string
}>
```
?

Reply:
Bringing in a new typing library would probably be a good case for a github discussion between more people. I will go ahead and just approve as is, but I do think it is worth having more conversation when we have time. Thanks for the explanations here though, I do appreciate it!